src/tpm.rs

```diff
@@ -26,50 +28,114 @@ pub const EXIT_SUCCESS: i32 = 0;
 
 static EMPTYMASK: &'static str = "1";
 
-/*
- * tpm data struct for tpmdata.json file IO
- */
-#[derive(Serialize, Deserialize, Debug)]
-struct TpmData {
-    aik_pw: String,
-    ek: String,
-    owner_pw: String,
-    aik_handle: String,
-    aikmod: String,
-    aikpriv: String,
-    aik: String,
-}
-
 /***************************************************************
 ftpm_initialize.py
 Following are function from tpm_initialize.py program      
 *****************************************************************/
 
 /*
-input: content key in tpmdata
-return: deserlized json object
-getting the tpm data struct and convert it to a json value object to access to a particular field inside the tpm data file
-*/
-fn get_tpm_metadata(content_key: String) -> Option<String> {
-    let t_data = read_tpm_data().unwrap();
-    let t_data_json_value = serde_json::to_value(t_data).unwrap();
-    Some(t_data_json_value[content_key].to_string())
+ * Input: content key in tpmdata
+ * Return: Result wrap value string or error code -1
+ *
+ * Getting the tpm data struct and convert it to a json value object to
+ * retrive a particular value by the given key inside the tpm data.
+ */
+fn get_tpm_metadata_content(key: &str) -> Result<Option<String>, i32> {
+    let tpm_data = match read_tpm_data() {
+        Ok(data) => data,
+        Err(e) => {
+            error!("Fail to read tpmdata.json, error: [{}].", e);
+            return Err(-1);
+        }
+    };
+    let remove: &[_] = &['"', ' ', '/'];
+    match tpm_data.get(key) {
+        Some(content) => Ok(Some(
+            content
+                .as_str()
+                .unwrap()
+                .to_string()
+                .trim_matches(remove)
+                .to_string(),
+        )),
+        None => {
+            debug!("Content field [{}] doesn't exist.", key);
+            Ok(None)
```

Comment:
Should be an `Err`, yeah?

Reply:
Changed to Err for a little simple implementation.